### PR TITLE
HEKClient search method now returns table instead of list of dict

### LINKED ISSUE #2740: We should return an astropy table from the HEK search method
The list of dicts is really ugly in comparison to an astropy table.

### PR DESCRIPTION
**Pull request by souravghosh97**
HEKClient.search() method returns table instead of list of dict
Closes #2740

### CODE REVIEW

**Comment by nabobalis** [comment]
Thanks @souravghosh97, do you have an example of what the output looks like now?

**Comment by souravghosh97** [comment]
```
>>> from sunpy.net import hek
>>> client = hek.HEKClient()  
>>> tstart = '2011/08/09 07:23:56'
>>> tend = '2011/08/09 12:40:29'
>>> event_type = 'FL'
>>> result = client.search(hek.attrs.Time(tstart,tend),hek.attrs.EventType(event_type))
>>> result
<HEKTable length=19>
         SOL_standard          active ar_compactnesscls ar_mcintoshcls ar_mtwilsoncls ar_noaaclass ar_noaanum ... skel_chaincode skel_curvature skel_nsteps skel_startc1 skel_startc2  sum_overlap_scores 
            str30               str4         str1            str1           str1          str1       object   ...      str1          object        object      object       object           str20        
------------------------------ ------ ----------------- -------------- -------------- ------------ ---------- ... -------------- -------------- ----------- ------------ ------------ --------------------
SOL2011-08-08T01:30:04L247C075   true                                                                    None ...                          None        None         None         None   12.588887041198964
SOL2011-08-08T01:30:04L247C075   true                                                                    None ...                          None        None         None         None   12.588887041198964
SOL2011-08-08T01:30:04L247C075   true                                                                    None ...                          None        None         None         None   12.588887041198964
SOL2011-08-09T01:40:04L230C084   true                                                                    None ...                          None        None         None         None  37.4574863857972034
SOL2011-08-09T02:30:04L319C077   true                                                                    None ...                          None        None         None         None  13.6403551847007094
SOL2011-08-09T02:30:04L319C077   true                                                                    None ...                          None        None         None         None  13.6403551847007094
SOL2011-08-09T02:30:04L319C077   true                                                                    None ...                          None        None         None         None  13.6403551847007094
SOL2011-08-09T07:19:00L296C075   true                                                                       0 ...                          None        None         None         None                    1
SOL2011-08-09T07:19:00L227C090   true                                                                   11263 ...                          None        None         None         None                    0
SOL2011-08-09T07:22:38L305C073   true                                                                    None ...                          None        None         None         None 0.283032539594587795
SOL2011-08-09T07:22:44L305C073   true                                                                    None ...                          None        None         None         None                    1
SOL2011-08-09T07:48:00L296C076   true                                                                       0 ...                          None        None         None         None                    0
SOL2011-08-09T07:48:00L296C073   true                                                                   11263 ...                          None        None         None         None                    0
SOL2011-08-09T07:55:59L305C073   true                                                                    None ...                          None        None         None         None                    0
SOL2011-08-09T07:59:49L300C077   true                                                                    None ...                          None        None         None         None                    0
SOL2011-08-09T08:00:03L305C073   true                                                                    None ...                          None        None         None         None                    0
SOL2011-08-09T08:00:20L305C073   true                                                                    None ...                          None        None         None         None                    0
SOL2011-08-09T08:00:53L305C073   true                                                                    None ...                          None        None         None         None                    0
SOL2011-08-09T08:01:21L300C077   true                                                                    None ...                          None        None         None         None                    0
>>> result[0]['event_peaktime']
'2011-08-09T11:55:00'
>>> result['event_peaktime']
<HEKColumn name='event_peaktime' dtype='str19' length=19>
2011-08-09T11:55:00
2011-08-09T11:55:00
2011-08-09T11:55:00
2011-08-09T08:07:00
2011-08-09T08:07:00
2011-08-09T08:07:00
2011-08-09T08:07:00
2011-08-09T07:23:00
2011-08-09T07:23:00
2011-08-09T08:07:38
2011-08-09T07:23:08
2011-08-09T08:05:00
2011-08-09T08:05:00
2011-08-09T08:06:23
2011-08-09T08:07:37
2011-08-09T08:06:51
2011-08-09T08:05:08
2011-08-09T08:07:05
2011-08-09T08:05:57
```

This is the output.

File: sunpy/net/hek/hek.py [hunk 1]
```diff
@@ -108,12 +109,25 @@ def _merge(self, responses):
...
+        elif table_item.__class__ == Row:
+            table_item.__class__ = HEKRow
+
+        return table_item
+
+class HEKColumn(Column):
+    pass
+
+class HEKRow(Row):
+    """Handles the response from the HEK.  Each HEKRow bject is a subclass
```
Comment (dpshelio) [inline sunpy/net/hek/hek.py]:
```suggestion
    """Handles the response from the HEK.  Each HEKRow object is a subclass
```

**Comment by souravghosh97** [comment]
I think the test should pass.

**Comment by nabobalis** [comment]
Thanks @souravghosh97

**Comment by irahulcse** [comment]
@nabobalis @dpshelio I would like to work on the issue #2759

**Comment by nabobalis** [comment]
Hi @irahulcse, I am not aware that there is anything else to do regarding this. 

I would suggest having a look at another open issue: https://github.com/sunpy/sunpy/issues

**Pull request merged**